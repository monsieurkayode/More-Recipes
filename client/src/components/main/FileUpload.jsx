import React from 'react';
import PropTypes from 'proptypes';
/**
 * Hanldes file selection for upload
 *
 * @function handleChange
 *
 * @param {function} handler
 *
 * @returns {object} selected file
 */
const handleChange = handler => ({ target: { files } }) =>
  handler(files.length ? { file: files[0], name: files[0].name } : {});

const FileUpload = ({
  input: { onChange, onBlur, value: omitValue, ...inputProps },
  meta: omitMeta,
  ...props
}) => (
  <div id="upload-btn">
    <label htmlFor="image">
      <input
        type="file"
        id="image"
        onChange={handleChange(onChange)}
        onBlur={handleChange(onBlur)}
        placeholder="Upload Image"
        {...inputProps}
        {...props}
      />
      <span
        className="btn"
      >
        Upload Image
      </span>
    </label>
  </div>
);

FileUpload.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func.isRequired,
    onBlur: PropTypes.func.isRequired
  }).isRequired,
  meta: PropTypes.shape({}).isRequired
};

export default FileUpload;
