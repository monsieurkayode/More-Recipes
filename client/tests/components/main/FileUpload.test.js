/* eslint-disable no-undef */
import React from 'react';

import FileUpload from '../../../src/components/main/FileUpload';

const setup = () => {
  const props = {
    input: {
      onChange: jest.fn(),
      onBlur: jest.fn(),
      value: null,
      meta: null
    }
  };

  const wrapper = shallow(<FileUpload {...props} />);

  return { wrapper };
};

const { wrapper } = setup();

describe('FileUpload', () => {
  it('should render correctly',
    (done) => {
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
});
