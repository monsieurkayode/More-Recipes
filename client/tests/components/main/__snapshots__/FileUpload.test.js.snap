// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload should render correctly 1`] = `
<div
  id="upload-btn"
>
  <label
    htmlFor="image"
  >
    <input
      id="image"
      meta={null}
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Upload Image"
      type="file"
    />
    <span
      className="btn"
    >
      Upload Image
    </span>
  </label>
</div>
`;
